staging/src/k8s.io/apiserver/pkg/authentication/audunaware/audunaware.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audunaware
```

Comment:
Awkward package name. I could also see these going directly in the authenticator package (the audience helpers and function-based authenticator implementations are already in that package)

Reply:
It breaks an import cycle caused by request.AudiencesFrom. I'm open to suggestions.

```
n dependency graph:
    //pkg/kubeapiserver/options:go_default_library
    //pkg/cloudprovider/providers:go_default_library
    //pkg/cloudprovider/providers/aws:go_default_library
    //pkg/controller:go_default_library
    //pkg/serviceaccount:go_default_library
.-> //staging/src/k8s.io/apiserver/pkg/authentication/authenticator:go_default_library
|   //staging/src/k8s.io/apiserver/pkg/endpoints/request:go_default_library
`-- //staging/src/k8s.io/apiserver/pkg/authentication/authenticator:go_default_library
```